api/app/src/providers/fitbit.ts

```diff
@@ -332,4 +335,71 @@ export class Fitbit extends Provider implements OAuth2 {
       }
     );
   }
+
+  async postAuth(token: string, userId?: string) {
+    console.log("Post auth started");
+    const accessToken = JSON.parse(token).access_token;
+
+    const scopes = JSON.parse(token).scope;
+
+    const subscriptionId = userId;
+
+    if (this.allRequiredScopesIncluded(scopes)) {
+      const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/apiSubscriptions/${subscriptionId}.json`;
+      this.createSubscription(subscriptionUrl, accessToken);
+      console.log("All scopes included. User subscribed to all Fitbit WH collectionTypes.");
+    } else {
+      if (scopes.includes("activity")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/activities/apiSubscriptions/${subscriptionId}activities.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+      if (scopes.includes("nutrition")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/foods/apiSubscriptions/${subscriptionId}foods.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+      if (scopes.includes("weight")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/body/apiSubscriptions/${subscriptionId}body.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+      if (scopes.includes("sleep")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/sleep/apiSubscriptions/${subscriptionId}sleep.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+    }
```

Comment:
There's a lot of duplication here, maybe we could:
- create a function that returns the URL based on a parameter
- build an array of urls to subscribe to and afterward go through it, instead of calling `createSubscription` on each condition/if

Reply:
oh, absolutely!